backend/inspirehep/search/facets.py

```diff
@@ -5,15 +5,21 @@
 # inspirehep is free software; you can redistribute it and/or modify it under
 # the terms of the MIT License; see LICENSE file for more details.
 
-from elasticsearch_dsl.query import Q, Range
+from elasticsearch_dsl.query import Q
 from flask import current_app, request
 
 
 def range_author_count_filter(field):
-    """Range filter for returning record only with 1 <= authors <= 10."""
+    """Range filter for returning records within the corresponding range(s)."""
+
+    range_for_option = {
+        "Single author": {"gte": 1, "lte": 1},
+        "10 authors or less": {"gte": 1, "lte": 10},
+    }
 
     def inner(values):
-        return Range(**{field: {"gte": 1, "lte": 10}})
+        ranges = [Q("range", **{field: range_for_option[value]}) for value in values]
+        return Q("bool", must=ranges)
```

Comment:
I know you just copied what others have been doing, but it's better to use a `filter` rather than a `must` for aggregations (not only here, all aggregations), see https://www.elastic.co/guide/en/elasticsearch/reference/5.6/query-filter-context.html for information about the difference.